lib/rex/random_identifier/generator.rb

```diff
@@ -247,12 +264,14 @@ def generate(len = nil)
     # pick a random length within the limits
     len ||= rand(@opts[:min_length] .. (@opts[:max_length]))
 
-    ident = ""
+    ident = ''
 
     # XXX: Infinite loop if block returns only values we've already
     # generated.
     loop do
-      ident  = Rex::Text.rand_base(1, "", @opts[:first_char_set])
+      ident  = ''
```

Comment:
Last nit:

```suggestion
      ident  = +''
```

https://gist.github.com/fxn/bf4eed2505c76f4fca03ab48c43adc72

Reply:
We'll probably want to start getting rid of the string literal mutations as we seem them come up, or at least avoid introducing new ones if we can